packages/tendermint-rpc/src/client.ts

```diff
@@ -248,14 +249,43 @@ export class Client {
     };
   }
 
-  public async validators(height?: number): Promise<responses.ValidatorsResponse> {
+  public async validators(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
     const query: requests.ValidatorsRequest = {
       method: requests.Method.Validators,
-      params: { height: height },
+      params: params,
     };
     return this.doCall(query, this.p.encodeValidators, this.r.decodeValidators);
   }
 
+  public async validatorsAll(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
+    let page = params.page || 1;
+    const validators: responses.Validator[] = [];
+    let done = false;
+    let blockHeight = 0;
+
+    while (!done) {
+      const resp = await this.validators({
+        per_page: 50,
+        ...params,
+        page: page,
+      });
+      validators.push(...resp.validators);
+      blockHeight = resp.blockHeight;
```

Comment:
Okay, this is good to record, but what happens if a block passes while querying (and the validators changed even?

I would explicitly set the height before querying. (See above)